File: models/classes.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Classes = sequelize.define('User', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  id_user: {
    type: DataTypes.INTEGER,
    allowNull: false,
    unique: true,
  },
  nome: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  classe: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  subclasse: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  talentos: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  passivas: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  items: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  xp: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  nivel: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  alinhamento: {
    type: DataTypes.STRING,
    allowNull: true,
  }
}, {
  timestamps: true,  // Cria automaticamente createdAt e updatedAt
  tableName: 'classes',  // Nome da tabela no banco
});

module.exports = Classes;

File: models/classes.go
package models
import (
	"time"
)
type Classes struct {
	ID         uint   `gorm:"primaryKey;autoIncrement"`
	IDUser     uint   `gorm:"not null"`
	Nome       string
	Classe     string
	Subclasse  string
	Talentos   []int `gorm:"type:integer[]"`
	Passivas   []int `gorm:"type:integer[]"`
	Items      []int `gorm:"type:integer[]"`
	XP         string
	Nivel      string
	Alinhamento string
	CreatedAt time.Time `gorm:"autoCreateTime;not null"` // Campo auto gerado
	UpdatedAt time.Time `gorm:"autoUpdateTime;not null"`

}